### PR TITLE
Fixed an issue in solve sometimes causing a hang

### LINKED ISSUE #15415: solve hangs on list of equations
While working on #15414 I found an intermittent case where `solve` hangs. I am not sure if this is the same as the intermittent problem but I'm reporting here a case that reliably hangs.

I'm finding that solve hangs when asked to solve a particular equation but only when that equation is given as a length-1 list of equations. This is fine:
```julia
In [6]: solve(Eq(y + 3*x**2/2, y + 3*x), y)
Out[6]: []
```

But this hangs:
```python
solve([Eq(y + 3*x**2/2, y + 3*x)], y)
```

### PR DESCRIPTION
**Pull request by warnerjon12**
#### References to other Issues or PRs

Fixes #15415

#### Brief description of what is fixed or changed
Fixed an issue in solve with one element iterables (used to hang)

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/solvers/solvers.py [hunk 1]
```diff
@@ -907,6 +907,8 @@ def solve(f, *symbols, **flags):
 
     def _sympified_list(w):
         return list(map(sympify, w if iterable(w) else [w]))
+    # denest one-element iterables. See issue #15415
+    while iterable(f) and len(f) == 1: f = next(iter(f))
```
Comment (sidhantnagpal) [inline sympy/solvers/solvers.py]: Could this be handled using an if condition?

This also needs test(s) to be included in `test_solvers.py`.

**Comment by sidhantnagpal** [comment]
There are a lot of failures due to this change. You can test your code using `./bin/test` and `./bin/doctest` inside your local repo as well.

**Comment by warnerjon12** [comment]
Yeah, I opened a new issue about the inconsistent solve return types. I
would like to see some comments there.

**Comment by smichr** [comment]
> see some comments there.

I added a comment. The one-element list should not be denested.

**Comment by warnerjon12** [comment]
> > see some comments there.
> 
> I added a comment. The one-element list should not be denested.

Yes, it seemed to cause other problems. I have now applied a different fix in `_invert` that seemed to be causing the issue.

**Comment by warnerjon12** [comment]
I looks like the tests are going to pass, then we are good for merge

**Comment by smichr** [comment]
You should probably update the title to better reflect what was fixed.

**Comment by warnerjon12** [comment]
We shouldn't need to wait for the tests to finish, I only changed the commit message and merged with master.

**Pull request merged**